crates/shared/src/transport/http.rs

```diff
@@ -303,4 +309,29 @@ mod tests {
         )
         .is_err());
     }
+
+    #[tokio::test]
+    #[ignore]
+    async fn batch_call_success() {
+        let http = create_env_test_transport();
+        let web3 = Web3::new(http);
+        let request = web3.transport().prepare("eth_blockNumber", Vec::default());
+        let request2 = web3.transport().prepare("eth_chainId", Vec::default());
+        web3.transport()
+            .send_batch([request, request2])
+            .await
+            .unwrap();
+        let metric_storage =
+            TransportMetrics::instance(global_metrics::get_metric_storage_registry()).unwrap();
```

Comment:
One issue I see with this test is that other tests may influence its result (since it uses a global storage registry). This could lead to flaky tests when run with other code that increments HTTP metrics.

Since this is an ignored test, this isn't super critical. Maybe we can create a local registry that is used just for this test.